numba/tests/test_extending.py

```diff
@@ -1748,12 +1749,56 @@ def testcase(x):
             got = testcase_cached(123)
             self.assertEqual(got, expect)
 
-            self.assertEqual(
-                testcase_cached._cache_hits[testcase.signatures[0]], 1,
-            )
-            self.assertEqual(
-                testcase_cached._cache_misses[testcase.signatures[0]], 0,
+    @classmethod
+    def check_objmode_cache_ndarray(cls):
+        def do_this(a, b):
+            return np.sum(a + b)
+
+        def do_something(a, b):
+            return np.sum(a + b)
+
+        @overload(do_something)
+        def overload_do_something(a, b):
+            def _do_something_impl(a, b):
+                with objmode(y='float64'):
+                    y = do_this(a, b)
+                return y
+            return _do_something_impl
+
+        @njit(cache=True)
+        def test_caching():
+            a = np.arange(20)
+            b = np.arange(20)
+            return do_something(a, b)
+
+        got = test_caching()
+        expect = test_caching.py_func()
+
+        # Check result
+        if got != expect:
+            raise AssertionError("incorrect result")
```

Comment:
```suggestion
        self.assertPreciseEqual(got, expected)
```

Reply:
It's a function that cannot be bound to the instance of `TestCase` because `TestCase` instances cannot be pickled, which is needed for the IPC used later.